File: dotnet/backend/Controllers/RazorpayController.cs
using Microsoft.AspNetCore.Mvc;
using Etour_Backend_dotnet.DTO.Razorpay;
using Etour_Backend_dotnet.Services.Razorpay;

namespace Etour_Backend_dotnet.Controllers;

[ApiController]
[Route("api/razorpay")]
public class RazorpayController : ControllerBase
{
    private readonly IRazorpayService _razorpayService;

    public RazorpayController(IRazorpayService razorpayService)
    {
        _razorpayService = razorpayService;
    }

    /// <summary>
    /// Create a Razorpay order for payment
    /// </summary>
    [HttpPost("create-order")]
    public IActionResult CreateOrder([FromBody] RazorpayOrderRequestDTO request)
    {
        try
        {
            var response = _razorpayService.CreateOrder(request);
            return Ok(response);
        }
        catch (Exception ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }
}

File: java/backend/src/main/java/com/etour/app/razorpay/controller/RazorpayController.java
package com.etour.app.razorpay.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.etour.app.razorpay.dto.RazorpayOrderRequestDTO;
import com.etour.app.razorpay.dto.RazorpayOrderResponseDTO;
import com.etour.app.razorpay.service.RazorpayService;

@RestController
@RequestMapping("/api/razorpay")
public class RazorpayController {

    private final RazorpayService razorpayService;

    public RazorpayController(RazorpayService razorpayService) {
        this.razorpayService = razorpayService;
    }

    @PostMapping("/create-order")
    public ResponseEntity<RazorpayOrderResponseDTO> createOrder(
            @RequestBody RazorpayOrderRequestDTO request
    ) {
        return ResponseEntity.ok(
                razorpayService.createOrder(request)
        );
    }
}
